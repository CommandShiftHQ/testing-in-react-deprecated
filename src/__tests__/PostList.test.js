import React from "react";
import renderer from "react-test-renderer";
import PostList from "../components/PostList";

describe("PostList", () => {
  const validProps = {
    posts: [
      {
        id: 1,
        author: "test author",
        body: "test body",
        date: "test date",
        isPublished: true,
        tags: ["test tag1", "test tag2", "test tag3"],
        title: "test title",
       },
      {
        id: 2,
        author: "test author 2",
        body: "test body 2",
        date: "test date 2",
        isPublished: false,
        tags: ["test tag1-2", "test tag2-2", "test tag3-2"],
        title: "test title 2",
      },
    ],
  };


  test("Renders as expected", () => {
    const testRenderer = renderer
      .create(<PostList posts={validProps.posts} />)
      .toJSON();

    expect(testRenderer).toMatchSnapshot();
  });
});