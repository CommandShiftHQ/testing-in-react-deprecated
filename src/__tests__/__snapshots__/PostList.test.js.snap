// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList Renders as expected 1`] = `
<div
  className="postlist"
>
  <div
    className="post"
  >
    <div
      className="post-heading"
    >
      <h2>
        test title
      </h2>
      <p>
        test body
      </p>
    </div>
    <div
      className="post-counter"
    >
      <span>
        Upvotes: 
        0
      </span>
      <button
        onClick={[Function]}
        type="button"
        value="test title"
      >
        Upvote this
      </button>
    </div>
    <div
      className="post-author"
    >
      Author: 
      test author
    </div>
    <div
      className="post-date"
    >
      Published: 
      test date
    </div>
    <h3>
      Tags:
    </h3>
    <ul>
      <li>
        test tag1
      </li>
      <li>
        test tag2
      </li>
      <li>
        test tag3
      </li>
    </ul>
  </div>
  <div
    className="post"
  >
    <div
      className="post-heading"
    >
      <h2>
        test title 2
      </h2>
      <p>
        Coming soon!
      </p>
    </div>
    <div
      className="post-counter"
    >
      <span>
        Upvotes: 
        0
      </span>
      <button
        onClick={[Function]}
        type="button"
        value="test title 2"
      >
        Upvote this
      </button>
    </div>
    <div
      className="post-author"
    >
      Author: 
      test author 2
    </div>
    <div
      className="post-date"
    >
      Published: 
      test date 2
    </div>
    <h3>
      Tags:
    </h3>
    <ul>
      <li>
        test tag1-2
      </li>
      <li>
        test tag2-2
      </li>
      <li>
        test tag3-2
      </li>
    </ul>
  </div>
</div>
`;
